moveit_experimental/moveit_jog_arm/src/jog_calcs.cpp

```diff
@@ -795,6 +808,7 @@ Eigen::VectorXd JogCalcs::scaleJointCommand(const control_msgs::JointJog& comman
     }
     catch (const std::out_of_range& e)
     {
+      const std::lock_guard<std::mutex> lock(latest_state_mutex_);
```

Comment:
Isn't this already locked at Line 733?

Reply:
I don't think so, but this lock_guard is in a weird place... I need to go through the PR and map out how I used them to make sure I didn't create a deadlock.  I'll do that.

Reply:
I looked at this and it doesn't do that.  I'm not sure how the code can trace from 733 to here at all.